src/lib-components/index.ts

```diff
@@ -109,6 +111,7 @@ export interface TreesComponents {
   BaseInput: typeof BaseInput
   Checkbox: typeof Checkbox
   DateRangePicker: typeof DateRangePicker
+  DateTime: typeof DateTime
```

Comment:
This is the place to nit the component name.  `DateTime`, DateTimeInput`, `DateTimeLocal` all seemed reasonable to me.  I went with the shortest after finding out that the `<input type="datetime" />` in the HTML spec is obsolete.